_episodes/05-gh-actions.md

```diff
@@ -0,0 +1,196 @@
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8 pytest
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+    - name: Lint with flake8
+      run: |
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+    - name: Test with pytest
+      run: |
+        pytest example.py
+```
+
+
+Do the commit by pressing  the "Start Commit" button:
+
+
+<img src="{{ site.baseurl}}/img/gh_action_commit.png"/>
+
+---
+
+### Step 4: Verify that tests have been automatically run
+
+Observe in the repository how the test succeed. While it the test is
+done, the repository has yellow marker. This is replaced with a green
+check mark, once the test succeeds:
+
+<img src="{{ site.baseurl}}/img/green_check_mark.png"/>
+
+---
+
+### Step 5: Add a test which reveals a problem
+
+Update your local cloned repository by doing a `git pull` or a combined `git fetch`
+and a `git merge orgin/master`. Next uncomment the code in `example.py` under "step 5", commit, and push.
+Verify that the test suite now fails, by selecting the yellow mark at
+"Latest commit":
+
+---
+<img src="{{ site.baseurl}}/img/test_failed.png"/>
+
+
+### Step 6: Open an issue on GitHub
+
+Open a new issue on the broken test on GitHub.
+The plan is that your colleague will fix the issue.
+
+
+### Step 7: Fork and clone the repository of your colleague
+
+Now you know the drill.
+
+
+### Step 8: Fix the broken test
+
+After you have fixed the code,
+commit the following commit message `"restore function subtract; fixes #1"` (assuming that you try to fix issue number 1).
+
+Once the pull request is accepted/merged, this will autoclose the issue since GitHub will recognize the "fixes #1" in the commit message, see also
+[closing issues using keywords](https://help.github.com/articles/closing-issues-using-keywords/).
+
+Then push to your fork.
+
+
+### Step 9: File a pull request
+
+Then before accepting the pull request from your colleague, observe
+how GitHub Actions automatically tested the code.
+
+
+### Step 10: Accept the pull request
+
+Observe how accepting the pull request automatically closes the issue (provided
+the commit message contained the correct issue number).
+
+Discuss whether this is a useful feature. And if it is, why do you think is it useful?
+
+
+### Step 11: Discussion
+
+We discuss together about our experiences with this exercise.
+
+---
+{: .challenge}
+
+## Where to go from here
```

Comment:
Also here we can link to the "marketplace" and encourage participants to find an interesting workflow which is close to their language/problem.